pandas/tests/test_categorical.py

```diff
@@ -54,6 +54,24 @@ def test_getitem_listlike(self):
         expected = c[np.array([100000]).astype(np.int64)].codes
         self.assert_numpy_array_equal(result, expected)
 
+    def test_getitem_category_type(self):
+        # GH 14580
+        # test iloc() on Series with Categorical data
+
+        s = pd.Series([1, 2, 3]).astype('category')
+
+        # get slice
+        result = s.iloc[0:1]
+        self.assertEqual(type(result), type(s))
```

Comment:
can you use ``tm.assert_series_equal`` for these (IOW, construct the expected series)